ledger-api/grpc-definitions/com/daml/ledger/api/v2/reassignment.proto

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "ReassignmentOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Complete view of an on-ledger reassignment.
+message Reassignment {
+
+  // Assigned by the server. Useful for correlating logs.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string update_id = 1;
+
+  // The ID of the command which resulted in this reassignment. Missing for everyone except the submitting party on the submitting participant.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in reassignment command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // The absolute offset. The format of this field is described in ``ledger_offset.proto``.
+  // Required
+  string offset = 4;
+
+  oneof event {
+    UnassignedEvent unassigned_event = 5;
+    AssignedEvent assigned_event = 6;
+  }
+}
+
+// Records that a contract has been unassigned, and it becomes unusable on the source domain
+message UnassignedEvent {
+
+  // The ID of the reassignment. This needs to be used as an input for a assign ReassignmentCommand.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string reassignment_id = 1;
+
+  // The ID of the reassigned contract.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string contract_id = 2;
+
+  //  The template of the reassigned contract.
+  //  Required
+  string template_id = 3;
+
+  // The ID of the source domain
+  // Must be a valid domain ID
+  // Required
+  string source = 4;
+
+  // The ID of the target domain
+  // Must be a valid domain ID
+  // Required
+  string target = 5;
+
+  // Party on whose behalf the reassignment command was executed.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string submitter = 6;
+
+  // Each corresponding assigned and unassigned event has the same reassignment_counter. This monotonically increases
+  // with each unassign command for the same contract. Creation of the contract corresponds to reassignment_counter
+  // equals zero.
+  // Required
+  uint64 reassignment_counter = 7;
+
+  // assignment exclusivity
+  // Before this time (measured on the target domain), only the submitter of the unassignment can initiate the assignment
+  // Missing for everyone except on the submitting participant
```

Comment:
Current implementation: not sure that it is true.
Goal: do we really want that. It is interesting for other participants to know when they try to submit the transfer-in, no?

Reply:
I have no strong opinion, sounds logical. @moritzkiefer-da what do you think?

Reply:
I think we should keep it. It's information that you need as a client because otherwise you cannot figure out when you need to submit your transfer.